### DISCUSSION TITLE
Initial Borg backup fails repeatedly

### DISCUSSION THREAD

**Discussion by FastlaneX** [nextcloud/all-in-one#5077] [Questions] [answered]
Hi at first thanks for this nice all-in-one Nextcloud solution.

I am facing problems with creating the nextcloud initial Borg backup. I previously had nextcloud aio running on a pi4 with two external HDDs in an USB dock without any problems (also borg working fine). Now I am running the same setup on a barebone N100 mini PC with 2 TB internal M2 SSD and the external USB HDD Mount with 2x2TB HDD running on ubuntu server 24.04 LTS.

Actually i am using SSD for containers, volumes etc. and also for nextcloud-datadir.
External HDDs are btrfs formatted, just using one of them for backup.

I rsync'ed all nextcloud volumes used with the pi to the SSD of the mini PC and created the nextcloud mastercontainer with the same compose file as used with the pi (using portainer). Everything is working fine and i can access all my NC apps and Data. Everything except borg.

Everytime i start borg it begins to create the archive and backing up data. Sometimes just hanging up on copying data to the archive (sometimes at 40Gb sometimes at 180Gb.. so on ) and sometimes exiting with an error.

example:

>B O 131.29 GB C 126.83 GB D 76921 N nextcloud_aio_v...0220915_160106.jpg
>cript.sh: line 273:    43 Killed                  borg create "${BORG_OPTS[@]}" "${BORG_EXCLUDE[@]}" >"$BORG_BACKUP_DIRECTORY::$CURRENT_DATE-nextcloud-aio" "/nextcloud_aio_volumes/"
>2024-07-31T20:51:19.293449087Z Killed stale lock nextcloud-aio-borgbackup.43-0.
>2024-07-31T20:51:19.395057422Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-07-31T20:51:19.395537226Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-07-31T20:51:19.488631641Z Killed stale lock nextcloud-aio-borgbackup.43-0.
>2024-07-31T20:51:19.590042695Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-07-31T20:51:19.590415653Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-07-31T20:51:19.594878901Z Archive 20240731_202552-nextcloud-aio not found (1/1).
>2024-07-31T20:51:19.594911478Z ------------------------------------------------------------------------------
>2024-07-31T20:51:19.594939790Z                        Original size      Compressed size    Deduplicated size
>2024-07-31T20:51:19.595016264Z Deleted data:                    0 B                  0 B                  0 B
>2024-07-31T20:51:19.595040257Z All archives:                    0 B                  0 B                  0 B
>2024-07-31T20:51:19.595080236Z 
>2024-07-31T20:51:19.595087450Z                        Unique chunks         Total chunks
>2024-07-31T20:51:19.595090267Z Chunk index:                       0                    0
>2024-07-31T20:51:19.595092544Z ------------------------------------------------------------------------------
>2024-07-31T20:51:19.736715903Z Backup failed!
>2024-07-31T20:51:19.736742808Z You might want to check the backup integrity via the AIO interface.
>2024-07-31T20:51:19.736746113Z Deleting borg.config file so that you can choose a different location for the backup.

I tried to create a backup also at the SSD at "/media/temp" with the same results multiple times. I checked the HDD with smartmontools resulting in a very good condition. To be on the safe side regarding to permissions i also chown and chmod the datair to 33:33 and 750.
I have seen similar problems here for discussion not coming to a helpful conclusion fixing my problem.

Maybe im missing something here and someone could help me. I appreciate every hint.

regards

**Comment by szaimen** [comment]
Hi, can you delete the nextcloud_aio_backup_cache volume and check if that helps?

**Reply by FastlaneX** [reply]
Thanks for your reply.
I will try that, and let you know.

**Reply by FastlaneX** [reply]
The backup is still failing after deleting cache volume and borg container. Even with the fresh container + volume it's going to crash, but now with different errors.

 >O 54.34 GB C 53.33 GB D 36937 N nextcloud_aio_vo.../20150516_005120.mp4
>thon error: Segmentation fault
>2024-08-01T18:26:26.613962713Z 
>2024-08-01T18:26:26.613965885Z Current thread 0x000078d7c9cb4b28 (most recent call first):
>2024-08-01T18:26:26.613968013Z   File "/usr/lib/python3.12/site-packages/borg/crypto/key.py", line 444 in encrypt
>2024-08-01T18:26:26.613969976Z   File "/usr/lib/python3.12/site-packages/borg/cache.py", line 950 in add_chunk
>2024-08-01T18:26:26.613973799Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1265 in chunk_processor
>2024-08-01T18:26:26.613975859Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1277 in process_file_chunks
>2024-08-01T18:26:26.613977750Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1455 in process_file
>2024-08-01T18:26:26.613979633Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 695 in _process_any
>2024-08-01T18:26:26.613981496Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 793 in _rec_walk
>2024-08-01T18:26:26.613983279Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613985040Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613986774Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613988504Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613990244Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613991967Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613993691Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
>2024-08-01T18:26:26.613995438Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 615 in create_inner
>2024-08-01T18:26:26.613997228Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 682 in do_create
>2024-08-01T18:26:26.614000006Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 190 in wrapper
>2024-08-01T18:26:26.614001873Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 5321 in run
>2024-08-01T18:26:26.614003824Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 5401 in main
>2024-08-01T18:26:26.614005631Z   File "/usr/bin/borg", line 33 in <module>
>2024-08-01T18:26:26.614119006Z 
>2024-08-01T18:26:26.614158096Z Extension modules: borg.crypto.low_level, msgpack._cmsgpack, borg.chunker, >borg.algorithms.checksums, borg.platform.posix, borg.platform.syncfilerange, borg.platform.linux, borg.hashindex, borg.item, >borg.compress, llfuse (total: 11)
>2024-08-01T18:26:30.863084380Z /backupscript.sh: line 273:    43 Segmentation fault      (core dumped) borg create >"${BORG_OPTS[@]}" "${BORG_EXCLUDE[@]}" "$BORG_BACKUP_DIRECTORY::$CURRENT_DATE-nextcloud-aio" >"/nextcloud_aio_volumes/"
>2024-08-01T18:26:30.863084538Z Deleting the failed backup archive...
>2024-08-01T18:26:31.382191292Z Killed stale lock nextcloud-aio-borgbackup.43-0.
>2024-08-01T18:26:31.483845963Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-08-01T18:26:31.484326871Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-08-01T18:26:32.490625114Z Killed stale lock nextcloud-aio-borgbackup.43-0.
>2024-08-01T18:26:32.592107248Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-08-01T18:26:32.592544036Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
>2024-08-01T18:26:32.597193041Z Archive 20240801_180948-nextcloud-aio not found (1/1).
>2024-08-01T18:26:32.597246853Z ------------------------------------------------------------------------------
>2024-08-01T18:26:32.597253127Z                        Original size      Compressed size    Deduplicated size
>2024-08-01T18:26:32.597271991Z Deleted data:                    0 B                  0 B                  0 B
>2024-08-01T18:26:32.597292368Z All archives:                    0 B                  0 B                  0 B
>2024-08-01T18:26:32.597325163Z 
>2024-08-01T18:26:32.597393731Z                        Unique chunks         Total chunks
>2024-08-01T18:26:32.597400583Z Chunk index:                       0                    0
>2024-08-01T18:26:32.597403307Z ------------------------------------------------------------------------------
>2024-08-01T18:26:34.417392808Z Backup failed!
>2024-08-01T18:26:34.417429675Z You might want to check the backup integrity via the AIO interface.
>2024-08-01T18:26:34.417433207Z Deleting borg.config file so that you can choose a different location for the backup.

seems to be something with the backup script?
Can you tell me what this thon error and the segmentation is about?

thanks!

**Reply by FastlaneX** [reply]
I googled a bit around and came to the conclusion, that it maybe is something about the memory?

NEXTCLOUD_MEMORY_LIMIT=4096M

should increase it to 4Gig for PHP right?
Does this environment variable also adjust the memory for borg? The other question is if borg has a memory limitation? 

I have 32G available.

Am I on the right path? 🙏

**Reply by szaimen** [reply]
Hm... Can you run the following commands
```
sudo docker rm nextcloud-aio-borgbackup
sudo docker image prune -af
sudo docker volume rm nextcloud_aio_backup_cache
```
and afterwards try again?

**Reply by FastlaneX** [reply]
I already did that, just not pruned the image. Will try with a fresh image.

**Reply by FastlaneX** [reply]
Unfortnately also ended in an error, nw just with some numbers in the "deleted data section" of the error instead of 0. Also showing segmentation fault at the same line in the backupscript as before. It almost fully backed up it were just some GBs left to complete the backp.

> B O 218.08 GB C 211.82 GB D 53845 N nextcloud_aio_v...0230920_211628.mp4
> thon error: Segmentation fault
> 2024-08-02T14:29:10.418072572Z 
> 2024-08-02T14:29:10.418076284Z Current thread 0x00007e48ea587b28 (most recent call first):
> 2024-08-02T14:29:10.418078510Z   File "/usr/lib/python3.12/site-packages/borg/crypto/key.py", line 400 in id_hash
> 2024-08-02T14:29:10.418080859Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1204 in cached_hash
> 2024-08-02T14:29:10.418082974Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1264 in chunk_processor
> 2024-08-02T14:29:10.418084924Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1277 in process_file_chunks
> 2024-08-02T14:29:10.418086834Z   File "/usr/lib/python3.12/site-packages/borg/archive.py", line 1455 in process_file
> 2024-08-02T14:29:10.418088695Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 695 in _process_any
> 2024-08-02T14:29:10.418090719Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 793 in _rec_walk
> 2024-08-02T14:29:10.418092587Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418094609Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418096529Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418098423Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418100267Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418102136Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418103991Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 830 in _rec_walk
> 2024-08-02T14:29:10.418105845Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 615 in create_inner
> 2024-08-02T14:29:10.418107722Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 682 in do_create
> 2024-08-02T14:29:10.418109571Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 190 in wrapper
> 2024-08-02T14:29:10.418111407Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 5321 in run
> 2024-08-02T14:29:10.418115611Z   File "/usr/lib/python3.12/site-packages/borg/archiver.py", line 5401 in main
> 2024-08-02T14:29:10.418117850Z   File "/usr/bin/borg", line 33 in <module>
> 2024-08-02T14:29:10.418282062Z 
> 2024-08-02T14:29:10.418296423Z Extension modules: borg.crypto.low_level, msgpack._cmsgpack, borg.chunker, borg.algorithms.checksums, borg.platform.posix, borg.platform.syncfilerange, borg.platform.linux, borg.hashindex, borg.item, borg.compress, llfuse (total: 11)
> 2024-08-02T14:29:10.675355323Z /backupscript.sh: line 273:    43 Segmentation fault      (core dumped) borg create "${BORG_OPTS[@]}" "${BORG_EXCLUDE[@]}" "$BORG_BACKUP_DIRECTORY::$CURRENT_DATE-nextcloud-aio" "/nextcloud_aio_volumes/"
> 2024-08-02T14:29:10.675355609Z Deleting the failed backup archive...
> 2024-08-02T14:29:11.208954431Z Killed stale lock nextcloud-aio-borgbackup.43-0.
> 2024-08-02T14:29:11.310563531Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
> 2024-08-02T14:29:11.311077015Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
> 2024-08-02T14:29:11.647394324Z Killed stale lock nextcloud-aio-borgbackup.43-0.
> 2024-08-02T14:29:11.748670708Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
> 2024-08-02T14:29:11.749138547Z Removed stale exclusive roster lock for host nextcloud-aio-borgbackup pid 43 thread 0.
> 2024-08-02T14:29:11.801022777Z Archive 20240802_132401-nextcloud-aio not found (1/1).
> 2024-08-02T14:29:11.810604345Z ------------------------------------------------------------------------------
> 2024-08-02T14:29:11.810629722Z                        Original size      Compressed size    Deduplicated size
> 2024-08-02T14:29:11.810633226Z Deleted data:                    0 B                  0 B                  0 B
> 2024-08-02T14:29:11.810635409Z All archives:               33.83 MB              7.61 MB            195.73 GB
> 2024-08-02T14:29:11.810639706Z 
> 2024-08-02T14:29:11.810641820Z                        Unique chunks         Total chunks
> 2024-08-02T14:29:11.810643834Z Chunk index:                  122940               127874
> 2024-08-02T14:29:11.810645776Z ------------------------------------------------------------------------------
> 2024-08-02T14:29:14.413644612Z Backup failed!
> 2024-08-02T14:29:14.413680026Z You might want to check the backup integrity via the AIO interface.
> 2024-08-02T14:29:14.413683546Z Deleting borg.config file so that you can choose a different location for the backup.

I don't get why it fails :(

I will try again on internal SSD.

**Reply by szaimen** [reply, marked as answer]
Honestly this looks like a bug in borg. If changing to the internal SSD also does not work, you might want to open a bug report at https://github.com/borgbackup/borg/

**Reply by FastlaneX** [reply]
Thanks for your time and your help.

If it fails again, i will open a bug report at borgbackup.

Have a nice weekend :)

**Reply by szaimen** [reply]
> Have a nice weekend :)

Thanks, you too! :)
